flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/batch/BatchExecSink.java

```diff
@@ -65,6 +65,25 @@
         },
         minPlanVersion = FlinkVersion.v2_0,
         minStateVersion = FlinkVersion.v2_0)
+// Version 2: Fixed the data type used for creating constraint enforcer and sink upsert
+// materializer. Since this version the sink works correctly with persisted metadata columns.
+// We introduced a new version, because statements that were never rolling back to a value from
+// state could run succesfully. We allow those jobs to be upgraded. Without a new versions such jobs
+// would fail on restore, because the state serializer would differ
+@ExecNodeMetadata(
```

Comment:
Batch version can stay at v1. There is no upgrade story that would interrupt a running batch job. I did it similarly for scan_v2, where I just updated v1 to the new behavior.